### DISCUSSION TITLE
TypeRig: Ideas and suggestions

### DISCUSSION THREAD

**Discussion by kateliev** [kateliev/TypeRig#47] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Wirdek** [comment]
Hello Vassil,
I have a question: Is there a way to automatically select nodes based on their type?
i.e. "select all sharp or smooth nodes at once"  
Either in FL (i did not find anything like that in it) or via TypeRig?

If selecting "sharp/corner", there could also be a filter for the angle (select right, acute or obtuse corners, for example...)

My goal (in my actual project) is to automatically select all sharp/corner nodes and then apply a "smart corner" to them?
(This could be a huge timesaver as my design is quite complicated)
Is there a way?

If not - do you find it interesting and perhaps worthy to implement?

Thanks in advance,
Vit

**Comment by kateliev** [comment]
Hey, Wirdek, interesting idea will think about it. Should be an easy thing - as long as sharp, smooth is concerned ... for filtering - well could be done as well. Give me some time, will try to squeeze this in my week schedule. If you do not hear from me until next week, feel free to remind me... i tend to forget stuff :)

**Comment by Wirdek** [comment]
Thank you!  :)

**Comment by Wirdek** [comment]
Hello Vassil,
I just want to ask if there is any progress?

Thanks,
(v)

**Comment by kateliev** [comment]
@Wirdek good timing! I am currently working on improving the TR Node panel. Will see what i can do :) Thanks for reminding me about this.

**Comment by Wirdek** [comment]
Thanks a lot!

**Comment by kateliev** [comment]
Hey @Wirdek it has been a while :) I just pushed a small update on `Outline panel` that will show you the node type, distance to previous and angle. It is an old an underused panel that enables you to view all the nodes in a simple table mode. Selecting a row will effectively select the node that it belongs to. Thus, with the new additions, you could sort the table according to preference - in your case smooth node type - and then just select those rows in the table and it will select those very same nodes in glyph window. You could also repeat the above for different angles and etc. Try it and tell me how it feels. I know that you expected something little bit more automated, but try this first and get back to me :)

![image](https://user-images.githubusercontent.com/10247582/213571031-09e025bc-493e-46cc-913d-da61a741855a.png)

_A side note: Currently the tool allows you to set coordinates numerically, but could be adapted to change node type... move nodes according to given distance and angle._

**Comment by Wirdek** [comment]
Hello Vassil,
I have just installed the update - I think it works just fine :)
Thanks a lot!

Are there any info-materials (videos, tutorials) regarding this new 3.0 version?

Have a nice day and thank you!
(v)
